vm/devices/storage/disk_layered/src/lib.rs

```diff
@@ -359,6 +373,81 @@ impl<T: LayerIo> DynLayer for T {
     }
 }
 
+trait DynAttachLayer: Send + Sync + Inspect {
+    fn attach(
+        self: Box<Self>,
+        lower_layer_metadata: Option<LowerLayerMetadata>,
+    ) -> Pin<Box<dyn Future<Output = anyhow::Result<AttachedDiskLayer>> + Send>>;
+}
+
+#[derive(Inspect)]
+#[inspect(transparent)]
+struct ErasedAttachLayer<T: AttachLayer>(T);
```

Comment:
Is this necessary? It seems like you could just have a blanket impl of `DynAttachLayer` on `T: AttachLayer` without the extra indirection.

Reply:
IIRC, I was running into some object-safety issues when I attempted that, due to the associated types. I'll take another pass now that things are actually compiling, and see if that was just a transient state.